Config: Full Parameter List: allow math expressions

IamPete1 opened the pull request:
This allows params to be entered as basic math expressions in the full list.

`1+2`, `5*10`, `8/9` ect

Brackets work, but not constants, powers or trig.

We could do more complex expressions if we were to add a math parser sub-module, but this gets us 99% of what anyone will use, and sub-modules are above by C# skill level.

meee1 commented:
mission planner already includes
MathParser.org-mXparser

in the "extlibs/utilities" project. so you could check out this as it may have better overall support for the wierd stuff

IamPete1 commented:
@meee1 even better, I have switched to that, things like `cos(pi)` work, `sqrt(25)` ect. While playing about I discovered you could get a NaN so I have also added handling for that case.

IamPete1 commented on GCSViews/ConfigurationView/ConfigRawParams.cs:
```diff
@@ -378,8 +379,16 @@ private void Params_CellValueChanged(object sender, DataGridViewCellEventArgs e)
+                if (float.IsNaN(newvalue) || float.IsInfinity(newvalue))
+                {
+                    Params.CellValueChanged -= Params_CellValueChanged;
+                    Params[e.ColumnIndex, e.RowIndex].Value = newvalue;
```
I put the new value, ie the `NaN`, so you can see your equation was wrong. Then again you could argue that it is better to leave it, the cell will go red thanks to the exception, and you can try and correct it. Maybe that would be nicer?

meee1 replied:
yeah, i could go either way. changing it to NaN means you would have to reload the screen to see the old value.

IamPete1 replied:
@meee1 I have changed to not over write the value. So no you get left with what you typed in with the red background. This mean you can correct the error in the equation. This is also consistent with what happens in current version if you enter  a none number.
